kernel/aster-nix/src/fs/ramfs/fs.rs

```diff
@@ -516,6 +516,8 @@ impl Inode for RamInode {
 
         let mut self_inode = self_inode.upgrade();
         self_inode.resize(new_size);
+        drop(self_inode);
+        let self_inode = self.0.read();
```

Comment:
This problem can also be resolved if implememting `downgrade` for `RwLockWriteGuard`, right? Like what [spin crate](https://docs.rs/spin/0.9.8/spin/rwlock/struct.RwLockWriteGuard.html#method.downgrade) does. By this way, we can avoid breaking atomicity.

Reply:
I agree with the `downgrade` method